### ISSUE TITLE
eAPI does not handle commands sent as unicode strings

### ISSUE THREAD

**Issue by jerearista** [arista-eosplus/pyeapi#137]
commands passed as unicode should be converted to ascii before being sent to eAPI.  
make_iterable may be a good place.  https://github.com/arista-eosplus/pyeapi/blob/8c396840de9cdfbf4ee83a78e15ef588b3017448/pyeapi/utils.py#L166

Similar code from elsewhere:
```
            for idx, command in enumerate(commands):
                if isinstance(command, unicode):
                   unicode commands[idx] = str(command)
```

**Comment by mharista** [comment]
I'm having trouble recreating this issue against my local veos instance. I'm running 4.17.2F and both string and unicode formats work for me.

>>> pp(cmds)
['interface Ethernet1', 'description string']
>>> pp(commands)
[u'interface Ethernet1', u'description change1']
>>> pp(node.config(cmds))
[{}, {}]
>>> pp(node.config(commands))
[{}, {}]

I verified commands worked on the veos instance as well.

**Comment by jerearista** [comment]
Fails in Python 2.7.10 but not in 3.6 with improved handling of unicode strings

```
(venv)$ python --version
Python 2.7.10
(venv)$ python
Python 2.7.10 (default, Jul 14 2015, 19:46:27) 
[GCC 4.2.1 Compatible Apple LLVM 6.0 (clang-600.0.39)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import pyeapi
>>> pyeapi.__version__
'0.8.0'
>>> node = pyeapi.connect_to('dut')
>>> node.enable('show hostname')
[{'command': 'show hostname', 'result': {u'hostname': u'localhost', u'fqdn': u'localhost'}, 'encoding': 'json'}]
>>> node.enable(u'show hostname')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/jere/src/network_validation/venv/lib/python2.7/site-packages/pyeapi/client.py", line 671, in enable
    **kwargs)
  File "/Users/jere/src/network_validation/venv/lib/python2.7/site-packages/pyeapi/client.py", line 729, in run_commands
    response = self._connection.execute(commands, encoding, **kwargs)
  File "/Users/jere/src/network_validation/venv/lib/python2.7/site-packages/pyeapi/eapilib.py", line 490, in execute
    response = self.send(request)
  File "/Users/jere/src/network_validation/venv/lib/python2.7/site-packages/pyeapi/eapilib.py", line 414, in send
    raise CommandError(code, msg, command_error=err, output=out)
pyeapi.eapilib.CommandError: Error [1002]: CLI command 2 of 2 's' failed: invalid command [ambiguous command (at token 0: 's')]
```

```
(venv3) $ python --version
Python 3.6.0
(venv3) $ python
Python 3.6.0 (default, Dec 24 2016, 08:02:28) 
[GCC 4.2.1 Compatible Apple LLVM 7.0.2 (clang-700.1.81)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import pyeapi
>>> pyeapi.__version__
'0.8.0'
>>> node = pyeapi.connect_to('dut')
>>> node.enable('show hostname')
[{'command': 'show hostname', 'result': {'fqdn': 'localhost', 'hostname': 'localhost'}, 'encoding': 'json'}]
>>> node.enable(u'show hostname')
[{'command': 'show hostname', 'result': {'fqdn': 'localhost', 'hostname': 'localhost'}, 'encoding': 'json'}]
>>> 
```

**Comment by mharista** [comment]
I must have done something stupid when attempting to recreate originally because I got it to work now. I also found the issue is specific to single unicode commands or lists containing unicode commands sent to the enable function. This is because the enable function loops through the list of commands and sends each command to run_commands individually and hence when the first unicode command in the list is sent as a single command to run_commands the issue arises.
